src/mesh.cpp

```diff
@@ -136,6 +138,41 @@ glm::vec3 Mesh::randomPoint()
     return ret;
 }
 
+uint32_t Mesh::greatestDistanceFromCenter(std::vector<MeshVertex>& vertices)
+{
+    if (vertices.empty()) {
+        return 0;
+    }
+
+
+    float sum_x = 0.f;
```

Comment:
maybe it would be possible to use `glm::vec3` instead? Not sure if you can initialize a `glm::vec3` from a `float[3]`.

But then it would look something like:
```
glm::vec3 sum{};
for(auto vertex : vertices) sum += glm::vec3(vertex.position);
auto average = sum / float(vertices.size());

for(auto vertex : vertices) {
    float distance = glm::length(glm::vec3(vertex.position) - average);
    ...
}
```

Reply:
I like that idea! That's where I started before changing it to try to match the style of the code above it, but I think it would be cleaner with `glm` structures/functions.